File: src/LorrisAnalyzer/DataWidgets/ScriptWidget/examples/graph.py
# Graph widget example.
import math

# you can comment out following line and create widget graph with title "graph"
graph = lorris.newWidget(WIDGET_GRAPH, "Graph", 700, 300, script.width + 20, 0);

# set graph scale
graph.setAxisScale(False, -1.2, 1.2); # for y axis
graph.setAxisScale(True, 0, 200); # for x axis

sin = None;
cos = None;
tg = None;

names = [ "sin", "cos", "tan" ] 

def onKeyPress(key):
	global names
	global cos
	global sin
	global tg

	if key == "s":
		if sin == None:
			sin = addCurve(0)
	elif key == "c":
		if cos == None:
			cos = addCurve(1)
	elif key == "t":
		if tg == None:
			tg = addCurve(2);
	elif key == "r": # Remove curves - they are indetified by name
		for name in names: # you can also use graph.removeAllCurves()
			graph.removeCurve(name);
		sin = cos = tg = None
	elif key == "m": # scroll graph so that it is at its maximal value of x axis
		graph.updateVisibleArea();

def addCurve(type):
	global names
	colors = [ "red", "blue", "green" ]
    
	# Add curve - first is name, second is color (as in html - name or hex #FFFFFF)
	c = graph.addCurve(names[type], colors[type]);

	x = 0;
	for i in range(500):
                # Add point to graph. first is index, second is value,
                # index is the value on x axis.
                # If you use same index twice, value if that index will be changed
		if type == 0:
			c.addPoint(i, math.sin(x))
		elif type == 1:
			c.addPoint(i, math.cos(x))
		elif type == 2:
			c.addPoint(i, math.tan(x))
		x += 0.1;
	return c;

File: src/LorrisAnalyzer/DataWidgets/ScriptWidget/examples/graph.js
// Graph widget example.
// you can comment out following line and create widget graph with title "graph"
var graph = newGraphWidget("Graph", 700, 300, script.width + 20, 0);

// set graph scale
graph.setAxisScale(false, -1.2, 1.2); // for y axis
graph.setAxisScale(true, 0, 200); // for x axis

var sin = null;
var cos = null;
var tg = null;

var names = new Array("sin", "cos", "tan");

// This function is called on key press in terminal.
// Param is string
function onKeyPress(key) {
	switch(key)
	{
		case "s":
			if(sin == null)
				sin = addCurve(0);
			break;
		case "c":
			if(cos == null)
				cos = addCurve(1);
			break;
		case "t":
			if(tg == null)
				tg = addCurve(2);
			break;
		case "r": // Remove curves - they are indetified by name
			for(var i = 0; i < names.length; ++i)
				graph.removeCurve(names[i]);
			sin = null;
			cos = null;
			tg = null;
			break;
		case "m": // scroll graph so that it is at its maximal value of x axis
			graph.updateVisibleArea();
			break;
	}	
}

function addCurve(type)
{
	var colors = new Array("red", "blue", "green");
    
	// Add curve - first is name, second is color (as in html - name or hex #FFFFFF)
	var c = graph.addCurve(names[type], colors[type]);

	var x = 0;
	for(var i = 0; i < 500; ++i)
	{
		// Add point to graph. first is index, second is value
		// index IS NOT value on x axis, but data will be ordered by this index
		// (point with idx 10 will be always before point with idx 50, even is it was inserted after)
        // if you use same index twice, value if that index will be changed
		switch(type)
		{
			case 0: c.addPoint(i, Math.sin(x)); break; 
			case 1: c.addPoint(i, Math.cos(x)); break;
			case 2: c.addPoint(i, Math.tan(x)); break;
		}
		x += 0.1;
	}
	return c;
}